### PR TITLE
feat: Deprecation DEPPS5: Config option `allowClientClassCreation` defaults to `false`

### LINKED ISSUE #8841: Remove deprecation DEPPS5: Config option `allowClientClassCreation` defaults to `false`
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Feature / Enhancement Description

Remove deprecation DEPPS5: Config option `allowClientClassCreation` defaults to `false`

### References 

https://github.com/parse-community/parse-server/pull/7925

### PR DESCRIPTION
**Pull request by onurhanife**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: #8841 

## Approach
Deprecation DEPPS5 removed

## Tasks

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

File: DEPRECATIONS.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ The following is a list of deprecations, according to the [Deprecation Policy](h
...
 | DEPPS3 | Config option `enforcePrivateUsers` defaults to `true` | [#7319](https://github.com/parse-community/parse-server/pull/7319)   | 5.0.0 (2022)                    | 6.0.0 (2023)                    | removed            | -     |
 | DEPPS4 | Remove convenience method for http request `Parse.Cloud.httpRequest` | [#7589](https://github.com/parse-community/parse-server/pull/7589)   | 5.0.0 (2022)                    | 6.0.0 (2023)                    | removed            | -     |
-| DEPPS5 | Config option `allowClientClassCreation` defaults to `false` | [#7925](https://github.com/parse-community/parse-server/pull/7925)   | 5.3.0 (2022)                    | 7.0.0 (2024)                    | deprecated            | -     |
+| DEPPS5 | Config option `allowClientClassCreation` defaults to `false` | [#7925](https://github.com/parse-community/parse-server/pull/7925)   | 5.3.0 (2023)                    | 7.0.0 (2024)                    | removed            | -     |
```
Comment (mtrezza) [inline DEPRECATIONS.md]:
```suggestion
| DEPPS5 | Config option `allowClientClassCreation` defaults to `false` | [#7925](https://github.com/parse-community/parse-server/pull/7925)   | 5.3.0 (2022)                    | 7.0.0 (2024)                    | removed            | -     |
```

**Review by mtrezza — APPROVED** [review]

**Review by mtrezza — CHANGES_REQUESTED** [review]
Tests fail, likely because of the default value change; I believe the easiest fix here is:
- Set `allowClientClassCreation: true` in the test config so that all existing tests pass
- Add new test in which `allowClientClassCreation` is not set (with `reconfigureServer`)
- Add new test in which `allowClientClassCreation: false` is set (with `reconfigureServer`)

**Comment by onurhanife** [comment]
@mtrezza made an update here

File: spec/ParseUser.spec.js [hunk 2]
```diff
@@ -24,11 +24,6 @@ describe('allowExpiredAuthDataToken option', () => {
     expect(Config.get(Parse.applicationId).allowExpiredAuthDataToken).toBe(false);
   });
 
-  it('should default false', async () => {
```
Comment (mtrezza) [inline spec/ParseUser.spec.js]: This seems to be related to a different option, why was this removed?

**Review by mtrezza — CHANGES_REQUESTED** [review]
We need a test for the new default value, to ensure the default is as expected.

File: spec/ParseUser.spec.js (hunk 2 above)
Comment (onurhanife) [reply spec/ParseUser.spec.js]: sorry, I read it wrong. thought since its already true in test configuration, testing it to default value would be wrong. but anyway, like i said i set default to true on test config. so, do you have any suggestions on how I can test to ensure the default is as expected?

Comment (mtrezza) [reply spec/ParseUser.spec.js]: Well, first, the test you removed `should default false` is under describe `allowExpiredAuthDataToken` option, so this is a different, and I believe unrelated option, so this removal should be reverted. You ask how to test a default value - that is exactly what the test that you removed does, but for a different option. You can just copy and adapt that test and add it under the correct `describe` that is related to the `allowClientClassCreation` option.

Comment (onurhanife) [reply spec/ParseUser.spec.js]: Yes i know, i will revert that but, as you know, with the change of `allowClientClassCreation` default to false, most of the test fails. so to avoid that, i set `allowClientClassCreation:true` on tests `defaultConfiguration` settings. You can see that on commits aswell. So, since its set to true on tests configuration, i cannot test it as exactly the same way with the one that i removed incorrectly. So its default false but configured to true. 

In order to test the `allowClientClassCreation:false` change, we either need to give `allowClientClassCreation:true` to the entire test with reconfigureServer to avoid fails (hundreds of tests will need to be changed), or we need to access to the default value in another way, which is exactly what I'm asking. How can we look at the `allowClientClassCreation`'s default value of the server?

Comment (mtrezza) [reply spec/ParseUser.spec.js]: Did you try to call `reconfigureServer({ allowClientClassCreation: undefined });` to try to override the default config? If that doesn't work the we may need to adapt this logic to make it work:

https://github.com/parse-community/parse-server/blob/dbe8b72d8ab0c5f1ba191ef844a3ce5cf9fd6249/spec/helper.js#L162

Comment (onurhanife) [reply spec/ParseUser.spec.js]: @mtrezza while i writing a logic to get defaultvalues, i see that `allowClientClassCreation`'s default value is still true. i tested it via deleting `allowClientClassCreation:true` on tests defaultConfiguration. and to be sure, i set default value of `allowClientClassCreation` to false in alpha branch to see if this branch is broken somehow but setting it to false didnt work at alpha branch also.  And also i noticed that `allowExpiredAuthDataToken`'s default value test(which i removed incorrectly) also fails. which should be false according to definitions. I tested that on alpha branch and confirmed that something wrong with default values, maybe a merged pr broke that.

Comment (onurhanife) [reply spec/ParseUser.spec.js]: @mtrezza sorry there was no failure on tests just something wrong on local. just made an update can you check

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-alpha.25](https://github.com/parse-community/parse-server/releases/tag/7.0.0-alpha.25)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0](https://github.com/parse-community/parse-server/releases/tag/7.0.0)
